cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsInstanceConnector.java

```diff
@@ -131,25 +126,31 @@ public String getConsoleOutput(AuthenticatedContext authenticatedContext, CloudI
     )
     @Override
     public List<CloudVmInstanceStatus> reboot(AuthenticatedContext ac, List<CloudInstance> vms) {
-        AmazonEC2Client amazonEC2Client = awsClient.createAccess(new AwsCredentialView(ac.getCloudCredential()),
-                ac.getCloudContext().getLocation().getRegion().value());
-        List<CloudInstance> affectedVms = new ArrayList<>();
+        List<CloudVmInstanceStatus> rebootedVmsStatus = new ArrayList<>();
+
         try {
             if (!vms.isEmpty()) {
                 List<CloudVmInstanceStatus> statuses = check(ac, vms);
-                doReboot(affectedVms, amazonEC2Client, getStarted(statuses));
-                doStart(affectedVms, ac, getStopped(statuses));
-                logInvalidStatuses(getNotStoppedOrStarted(statuses));
+                doStop(new ArrayList<>(), ac, getStarted(statuses));
+                statuses = check(ac, vms);
+                logInvalidStatuses(getNotStopped(statuses));
+                rebootedVmsStatus = doStart(new ArrayList<>(), ac, getStopped(statuses));
+                logInvalidStatuses(getNotStarted(statuses));
```

Comment:
I think `logInvalidStatuses` would need some extra parameter which is also logged. Like "instances should be in `STOPPED` state" and here comes the list of instance ids and states. Otherwise we won't know if the log is coming from line 136 or 138.
Also `logInvalidStatuses` should log only one line, and not separate lines per instance. Could you fix that?

Reply:
I fixed both here. 

Now it will only log one line. Also there is a targetStatus parameter to log the expected status.

Below is a sample log message. 
`
Unable to reboot instance i-1 because of invalid status STARTED, instance i-2 because of invalid status STARTED. Instances should be in STOPPED state.`

Reply:
I have also squashed all my commit into one commit.